hail/python/hailtop/fs/router_fs.py

```diff
@@ -117,7 +117,7 @@ def isatty(self):
     def readable(self):
         return False
 
-    def readline(self, size=-1):
+    def readline(self, size: Optional[int] = -1):
```

Comment:
What does `size=None` mean?

Reply:
Python folks got back to me. This is actually intentional. Python supports None as a synonym of -1. It does not support this behavior on the TextIOBase subclass, so Python is internally inconsistent. 🤷